### PR TITLE
SWORD locales.d/locales.conf to json

### PR DESCRIPTION
**Pull request by zhuiks**
This script will generate `lib/languages.json` for quick language resolving in Ezra Bible App frontend.

It should be run with node 14+, because there is full node support of [Intl API](https://nodejs.org/dist/latest-v14.x/docs/api/intl.html) starting from v 14

### CODE REVIEW

**Comment by zhuiks** [comment]
It would be nice to run with Github actions, but I can't think about workflow should work with build and publish

**Comment by tobias-klein** [comment]
Oh, I didn't see this when commenting on the other PR in the Ezra Bible App repo. I'll check this out tomorrow.

File: scripts/sword_locales_to_json.mjs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  });
+}
+
+function toTitleCase(str) {
+  return str.slice(0, 1).toLocaleUpperCase() + str.slice(1);
+}
+
+function saveToJsonFile(data, filename) {
+  const jsonData = JSON.stringify(data);
+  // const jsonData = JSON.stringify(data, null, 2); // If pretty print is needed
```
Comment (tobias-klein) [inline scripts/sword_locales_to_json.mjs:247]: Would this also result in an output with linebreaks? I would prefer the pretty print version actually.

File: scripts/sword_locales_to_json.mjs [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+if (!hasFullICU) {
+  console.log(`No Intl API support detected for this node version. Script results will be limited`);
+}
+
+if (!isTesting()) {
+  indexedLangs = indexIso6393Langs();
+
+  const languages = parseSwordLocales();
+
+  const allLocales = await getEzraBibleAppLocales(EZRA_LOCALES_URL);
```
Comment (tobias-klein) [inline scripts/sword_locales_to_json.mjs:32]: I would prefer to keep node-sword-interface independent of Ezra Bible App. But I guess the requirement would be to maintain the supported locale list separately here, huh?

**Comment by tobias-klein** [comment]
> It should be run with node 14+, because there is full node support of [Intl API](https://nodejs.org/dist/latest-v14.x/docs/api/intl.html) starting from v 14

I think we should document that requirement in the README file and also provide a short instruction on the purpose and use of this script, would you mind doing that?

File: scripts/sword_locales_to_json.mjs (hunk 1 above)
Comment (zhuiks) [reply scripts/sword_locales_to_json.mjs:247]: Yes, it will be with the line breaks.

File: scripts/sword_locales_to_json.mjs (hunk 2 above)
Comment (zhuiks) [reply scripts/sword_locales_to_json.mjs:32]: Yes, "circle" dependencies are not good. Maybe this script should be back in Ezra Bible App. 
The other argument for this is the difference in Node versions.

File: scripts/sword_locales_to_json.mjs (hunk 1 above)
Comment (tobias-klein) [reply scripts/sword_locales_to_json.mjs:247]: Ok, then I would definitely suggest to switch to the pretty print version.

**Comment by zhuiks** [comment]
This is implemented on Ezra Bible App side

**Pull request closed without merging**